### PR TITLE
Update concourse

### PR DESCRIPTION
**Pull request by huwd**
Trello: https://trello.com/c/t6jxzyCY/229-dockerize-app-and-run-tests-within-concourse

## What
- Adds a single docker image we can use for all test builds

- Give resources shorter, more generic names. It is clear from within a
pipeline which service we are in, and long names make for things
bunching up on the UI, so generally simplify everything down to just
`git-master` or simple descriptions of what the resource actually is.

- Create a run quality check task, this should be able to take a
concourse test docker image with all the system level dependencies we
need (Redis/Postgres) and then provide all the commands needed to set up
and run the test suite + linting.

- Rearrange the triggers in the pipeline so quality checks run every
time we get a change to the master branch. These quality checks will
then be a dependency for deployments to staging and production. If they
fail, then we don't go any further.

## Why

Allows us great reassurance that any build heading down a pipeline to production passes our required quality checks.

## Follow up
After this, I'll put in a separate tidy up PR that will:
- Remove all travis config
- Remove the features Dockerfile
- Tidy up anything else we no longer need after this.

### CODE REVIEW

File: concourse/pipeline.yml [hunk 1]
```diff
@@ -5,39 +5,23 @@ resource_types:
...
-      pro: true
-
-  - name: govuk-coronavirus-business-volunteer-form-feature-tests
-    type: git
+  - name: git-master-gems-dockerfile
     icon: github-circle
+    type: git
     source:
-      uri: https://github.com/alphagov/govuk-coronavirus-business-volunteer-form
+      uri:  https://github.com/alphagov/govuk-coronavirus-business-volunteer-form
```
Comment (issyl0) [inline concourse/pipeline.yml]:
```suggestion
      uri: https://github.com/alphagov/govuk-coronavirus-business-volunteer-form
```

File: concourse/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#========================================================================================
+# This Dockerfile creates a container image which contains Ruby,
+# Google Chrome and Postgres, as well as the application source and bundled dependencies.
```
Comment (huwd) [inline concourse/Dockerfile]:
```suggestion
# Google Chrome, Redis and Postgres, as well as the application source and bundled dependencies.
```

File: concourse/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#
+# This is needed for running the feature tests in a container environment like Concourse.
+#========================================================================================
+FROM ruby:2.6.5-buster
+
+# Updates all debian dependencies
+# Adds google chrome to software sources, install chrome, install postgres
```
Comment (huwd) [inline concourse/Dockerfile]:
```suggestion
# Adds google chrome to software sources, install chrome, install postgres and redis
```

File: concourse/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+# clear up lists behind ourselves
+RUN apt-get update --fix-missing && apt-get -y upgrade \
+    && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
```
Comment (huwd) [inline concourse/Dockerfile:13]: Does this app still need chrome headless?
Rspec feature tests use it right?

File: concourse/pipeline.yml (hunk 1 above)
Comment (huwd) [inline concourse/pipeline.yml]:
```suggestion
      uri: https://github.com/alphagov/govuk-coronavirus-business-volunteer-form
```

File: concourse/pipeline.yml [hunk 5]
```diff
@@ -5,39 +5,23 @@ resource_types:
...
       branch: master
       paths:
+        - concourse/Dockerfile
         - Gemfile*
-        - .ruby-version
-        - spec/
```
Comment (huwd) [inline concourse/pipeline.yml:40]: Not sure we need to have conrouse monitor the `/spec` file for this, we'd only want to rebuild this for dependency or docker changes i think :thinking:

File: concourse/pipeline.yml [hunk 6]
```diff
@@ -49,42 +33,77 @@ resources:
...
+            dir: git-master
+            path: bash
+            args:
+              - -c
+              - |
+                set -eu
+                service postgresql start
+                su - postgres -c "psql -c \"create role root with createdb login password 'password';\""
+                service redis-server start
+                export TEST_DATABASE_URL="postgres://root:password@localhost:5432/coronavirus_business_volunteer_form_test"
```
Comment (huwd) [inline concourse/pipeline.yml:91]: Should reflect this stuff back on the find-support form.
I think the app thinks it's running in development over there.

**Comment by huwd** [comment]
Interesting Heroku errored, is that happening elsewhere?

File: concourse/pipeline.yml (hunk 5 above)
Comment (issyl0) [reply concourse/pipeline.yml:40]: If the Dockerfile to build the features image is in `concourse/` now then no, that's fine, we don't need `spec`. We probably _do_ want when `.ruby-version` changes, but with that would come a `Dockerfile` change to pull `ruby:2.6.6-buster` so I think that's fine to leave out.

Comment (huwd) [reply concourse/pipeline.yml:40]: :+1:

File: concourse/Dockerfile (hunk 4 above)
Comment (issyl0) [reply concourse/Dockerfile:13]: I believe that's right.

**Review by issyl0 — APPROVED** [review]

**Pull request merged**